modules/db/src/main/java/org/mule/module/db/internal/result/statement/StatementResultIterator.java

```diff
@@ -89,9 +89,11 @@ public boolean hasNext()
                     }
                 }
 
-                if (generatedKeys == null)
+                if (generatedKeys == null && !moveToNextResult() && statement.getUpdateCount() == NO_UPDATE_COUNT)
                 {
-                    moveToNextResult();
+                    updateCount = statement.getUpdateCount();
```

Comment:
You are calling statement.getUpdateCount() twice in the same read. I think this change the statement state, so the second time it maybe will read again.